packages/@react-spectrum/table/src/InsertionIndicator.tsx

```diff
@@ -0,0 +1,58 @@
+import {classNames} from '@react-spectrum/utils';
+import {ItemDropTarget} from '@react-types/shared';
+import React, {useCallback, useRef} from 'react';
+import styles from './table.css';
+import {useTableContext} from './TableView';
+import {useVisuallyHidden} from '@react-aria/visually-hidden';
+
+interface InsertionIndicatorProps {
+  target: ItemDropTarget,
+  rowProps: any
+}
+
+export default function InsertionIndicator(props: InsertionIndicatorProps) {
+  let {dropState, dragAndDropHooks, state} = useTableContext();
+  const {target, rowProps} = props;
+
+  let getKeyBefore = useCallback((key) => {
```

Comment:
Something to discuss with team, better to have getKeyBefore and getKeyAfter to pass into the hook? or would it be better to pass a filter function (much like the dom tree walker has)